Apsitvarkom.Api/Program.cs

```diff
@@ -1,13 +1,18 @@
 using Apsitvarkom.DataAccess;
 using Apsitvarkom.Models.DTO;
 using Apsitvarkom.Models.Mapping;
+using AutoMapper;
 using FluentValidation;
 
 var builder = WebApplication.CreateBuilder(args);
 
 // Add services to the container.
 
-builder.Services.AddScoped<IPollutedLocationDTORepository>(_ => PollutedLocationDTOFileRepository.FromFile("PollutedLocationsMock.json"));
+builder.Services.AddScoped<IPollutedLocationDTORepository>(_ => PollutedLocationDTOFileRepository.FromFile(new MapperConfiguration(cfg =>
+{
+    cfg.AddProfile<PollutedLocationProfile>();
+}).CreateMapper(), "PollutedLocationsMock.json"));
+
```

Comment:
```csharp
builder.Services.AddScoped<IPollutedLocationDTORepository>(serviceProvider =>
{
    var mapper = serviceProvider.GetRequiredService<IMapper>();
    return PollutedLocationDTOFileRepository.FromFile(mapper, "PollutedLocationsMock.json");
});
```

Reply:
`mapper` will be automatically added to dependency injection container by [`AddAutoMapper` method](https://github.com/vu-vibedosa/apsitvarkom-be/blob/dev/Apsitvarkom.Api/Program.cs#L17).

Reply:
Thanks, this makes sense. I was just confused to what would happen if in the future we would have more AutoMapper profiles - which one would be chosen? - That's the reason of me sticking with this implementation, that was way off of seeming to be ideal, but I was clear would work right. 

Fixed it. I'm also not sure if that was necessary but I also moved the statement below AddAutoMapper. After testing it, it seemed to work just fine both ways - is there any prefered one? Let me know if I should revert this.

Reply:
> if in the future we would have more AutoMapper profiles - which one would be chosen?

As I understand it, profiles are only used to group mappings. When you register automapper (call `AddAutoMapper`) to DI, it picks up all profiles and adds an instance of `IMapper` to the DI container – this instance knows all mappings (combined from all profiles). What would happen if you were to define two *mappings* (does not matter if they're in different *profiles* or not) from `Type1` to `Type2` I can't tell, I expect it to throw an error because it won't know which one to choose – but either way you should probably not end up in this situation.

> is there any prefered one?

The order should not matter – the services you register to DI get evaluated/created later (not at the time where you declare them). There's no preferred way AFAIK